### PR TITLE
alter tables that create unique keys that are too long to have row_format DYNAMIC

### PR DESCRIPTION
**Pull request by fabada**
@ENuge I found all of the tables that were creating unique keys that violated the max length of 767 bytes and altered the tables to have row_format=DYNAMIC. This, in addition to have innodb_file_format=barracuda and innodb_large_prefix=true in our mysql configs should change the restriction to be 3072 bytes, and thus we can run these migrations without errors.

### CODE REVIEW

File: pootle/apps/pootle_app/migrations/0015_auto__add_unique_directory_pootle_path.py [hunk 1]
```diff
@@ -12,10 +12,13 @@ class Migration(SchemaMigration):
 
     def forwards(self, orm):
         # Adding unique constraint on 'Directory', fields ['pootle_path']
+        # Set table row format to DYNAMIC to change unique key length constraint from 767 bytes to 3072
+        db.execute("ALTER TABLE `pootle_app_directory` ROW_FORMAT=DYNAMIC")
         db.create_unique(u'pootle_app_directory', ['pootle_path'])
 
     def backwards(self, orm):
         # Removing unique constraint on 'Directory', fields ['pootle_path']
+        db.execute("ALTER TABLE `pootle_app_directory` ROW_FORMAT=COMPACT")
```
Comment (ENuge) [inline pootle/apps/pootle_app/migrations/0015_auto__add_unique_directory_pootle_path.py]: Why is this needed? (Is it already dynamic or something, and that's bad to keep in general? I guess it's space inefficient.)

File: pootle/apps/pootle_project/migrations/0001_initial.py [hunk 2]
```diff
@@ -27,6 +27,8 @@ def forwards(self, orm):
             ('report_target', self.gf('django.db.models.fields.CharField')(max_length=512, blank=True)),
         ))
         db.send_create_signal('pootle_project', ['Project'])
+        db.execute("ALTER TABLE `pootle_app_project` ROW_FORMAT=DYNAMIC")
```
Comment (ENuge) [inline pootle/apps/pootle_project/migrations/0001_initial.py]: It's annoying you can't do that inline, but this works. :+1:

File: pootle/apps/pootle_app/migrations/0015_auto__add_unique_directory_pootle_path.py (hunk 1 above)
Comment (fabada) [inline pootle/apps/pootle_app/migrations/0015_auto__add_unique_directory_pootle_path.py]: It's essentially a rollback step. Not sure how much this is needed since I don't know how to rollback a migration done from `pootle setup`

Comment (ENuge) [inline pootle/apps/pootle_app/migrations/0015_auto__add_unique_directory_pootle_path.py]: Oh I see. Seems safer to keep in case someone else does end up using this to rollback, ++ for thinking about it.
